src/ui/public/visualize/visualize.html

```diff
@@ -1,4 +1,4 @@
-<div ng-if="vis.type.requiresSearch && esResp.hits.total === 0"
+<div ng-if="showNoResultsMessage(esResp.hits.total === 0)"
```

Comment:
Take the logic out of the view here. There is already logic in the function in the `visualize.js`